### PR TITLE
drop py<3.9

### PR DESCRIPTION
**Pull request by trim21**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -70,9 +70,6 @@ line-length = 92
 # See 'rust-format.el' and .dir-locals.el at the root of the project.
 
 [tool.ruff]
```
Comment (trim21) [inline pyproject.toml:72]: ruff can understand `project.requires-python`

**Comment by yagebu** [comment]
Why drop Python 3.9? I think it's good to support all officially supported Python versions - unless there's something we really need in 3.10+?

**Comment by trim21** [comment]
> Why drop Python 3.9? I think it's good to support all officially supported Python versions - unless there's something we really need in 3.10+?

not really IMO

https://github.com/beancount/beancount/pull/910#discussion_r1895804150

**Comment by trim21** [comment]
> Why drop Python 3.9? I think it's good to support all officially supported Python versions - unless there's something we really need in 3.10+?

OH, sorry. we do not need to drop 39. I'll add 39 back

**Comment by blais** [comment]
I think I've done this an hour ago more or less

**Pull request closed without merging**